src/averageAges.js

```diff
@@ -1,61 +1,61 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let filteredByCentury = people.filter(person =>
+    Math.ceil(person.died / 100) === century
+  );
+
+  if (century === undefined) {
+    filteredByCentury = people;
+  }
+
+  const filteredBySex = filteredByCentury.filter(person => person.sex === 'm');
+
+  const sumOfMenAges = filteredBySex.reduce((sum, men) =>
+    sum + (men.died - men.born), 0);
+
+  return sumOfMenAges / filteredBySex.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let filterByMotherStatus = people.filter(women =>
+    women.sex === 'f'
+    && people.some(person => person.mother === women.name)
+  );
+
+  if (withChildren === undefined) {
+    filterByMotherStatus = people.filter(women => women.sex === 'f');
+  }
```

Comment:
Due to the task description:
replace `if ()` statement with &&, || or ?:

Check all such cases